File: client/interactions.lua
ped = {}

local function deletePeds()
    if not pedSpawned then return end
    for _, v in pairs(ped) do
        DeletePed(v)
    end
    pedSpawned = false
end

CreateThread(function()
    if pedSpawned then return end
    
    for k, v in pairs(Config.Interact) do
        local current = type(v.ped) == "number" and v.ped or GetHashKey(v.ped)
        
        RequestModel(current)
        while not HasModelLoaded(current) do
            Wait(0)
        end
        
        ped[k] = CreatePed(0, current, v.coords.x, v.coords.y, v.coords.z-1, v.coords.w, false, false)
        TaskStartScenarioInPlace(ped[k], v.scenario, 0, true)
        FreezeEntityPosition(ped[k], true)
        SetEntityInvincible(ped[k], true)
        SetBlockingOfNonTemporaryEvents(ped[k], true)

    end
    pedSpawned = true

    for i, v in pairs(Config.Interact) do
        
        local interactionGroups = {} -- Initialize an empty table for groups
        for jobName, requiredGrade in pairs(v.jobs) do
            interactionGroups[jobName] = requiredGrade -- Add each job and its grade to the groups table
            
        end
        exports['qb-target']:AddBoxZone("gov.garage_" .. i, vector3(v.coords.x, v.coords.y, v.coords.z), 3.45, 3.35, {
            name = "gov.garage_" .. i,
            minZ = v.coords.z - 1.0,
            maxZ = v.coords.z + 1.0,
        }, {
            options = {
                {
                    type = "client",
                    icon = v.icon,
                    label = v.label,
                    job = interactionGroups,
                    action = function()
                        TriggerEvent("lenix_patrolvehicles:menu", v)
                    end,
                },
            },
            distance = v.distance,
        })
    end
end)

CreateThread(function()
    if GetResourceState(GetCurrentResourceName()) == 'stopping' then
        deletePeds()
    end
end)

File: client/interactions.js
for (const [key, value] of Object.entries(System)) {
    if (key === "_DEFAULT") continue
    for (const pedElement of value.PEDS.peds) {
        const zoneName = `${GetCurrentResourceName()}_${key}_${pedElement}`
        const interactions = {
            ...System._DEFAULT.INTERACTIONS,
            ...pedElement.INTERACTIONS,
            take: {
                ...System._DEFAULT.INTERACTIONS.take,
                ...pedElement.INTERACTIONS?.take
            },
            return: {
                ...System._DEFAULT.INTERACTIONS.return,
                ...pedElement.INTERACTIONS?.return
            },
            targets: {
                ...System._DEFAULT.INTERACTIONS.targets,
                ...pedElement.INTERACTIONS?.targets
            }
        }
        exports['qb-target'].AddBoxZone(
            zoneName,
            {
                x: pedElement.coords[0],
                y: pedElement.coords[1],
                z: pedElement.coords[2]
            },
            3.45,
            3.35,
            {
                name: zoneName,
                heading: pedElement.coords[3],
                debugPoly: interactions.debug,
                minZ: pedElement.coords[2] - 1.0,
                maxZ: pedElement.coords[2] + 1.0,
            }, 
            {
                options: [
                    {
                        type: "client",
                        icon: interactions.take.icon,
                        label: interactions.take.label,
                        job: interactions.targets.job,
                        gang: interactions.targets.gang,
                        action: function() {
                            Menu.main(key)
                        },
                    },
                ],
                distance: interactions.take.distance,
            }
        )
    }
}

function returnVehicle(key) {
    
}